homeassistant/components/hive/sensor.py

```diff
@@ -1,46 +1,103 @@
 """Support for the Hive sensors."""
+from collections.abc import Callable
+from dataclasses import dataclass
 from datetime import timedelta
 
-from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import TEMP_CELSIUS
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from . import HiveEntity
 from .const import DOMAIN
 
 PARALLEL_UPDATES = 0
 SCAN_INTERVAL = timedelta(seconds=15)
-DEVICETYPE = {
-    "Camera_Temp": {
-        "icon": "mdi:thermometer",
-        "unit": TEMP_CELSIUS,
-        "type": "temperature",
-    },
-    "Heating_Current_Temperature": {
-        "icon": "mdi:thermometer",
-        "unit": TEMP_CELSIUS,
-        "type": "temperature",
-    },
-    "Heating_Target_Temperature": {
-        "icon": "mdi:thermometer",
-        "unit": TEMP_CELSIUS,
-        "type": "temperature",
-    },
-    "Heating_State": {"icon": "mdi:radiator"},
-    "Heating_Mode": {"icon": "mdi:radiator"},
-    "Heating_Boost": {"icon": "mdi:radiator"},
-    "Hotwater_State": {"icon": "mdi:water-pump"},
-    "Hotwater_Mode": {"icon": "mdi:water-pump"},
-    "Hotwater_Boost": {"icon": "mdi:water-pump"},
-    "Mode": {"icon": "mdi:eye"},
-    "Battery": {"unit": " % ", "type": SensorDeviceClass.BATTERY},
-    "Availability": {"icon": "mdi:check-circle"},
-    "Connectivity": {"icon": "mdi:check-circle"},
-    "Power": {"type": SensorDeviceClass.ENERGY},
-}
+
+
+@dataclass
+class HiveSensorEntityDescription(SensorEntityDescription):
+    """Class describing Hive sensor entities."""
+
+    value: Callable = round
+
+
+SENSOR_TYPES: tuple[HiveSensorEntityDescription, ...] = (
+    HiveSensorEntityDescription(
+        key="Heating_Current_Temperature",
+        icon="mdi:thermometer",
+        device_class="temperature",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        entity_category=EntityCategory.DIAGNOSTIC,
```

Comment:
Should all these sensors really be diagnostic? Normally we only set secondary information as diagnostic, eg battery level, tamper state and similar.

Reply:
I suppose I need to understand how home assistant categorises its information. there is certain data within Hive like mode of a device which doesn't fit into the control entity of the device how should that be represented in Home assistant?

Reply:
What does the device mode mean?

Here's the docs:
https://developers.home-assistant.io/docs/core/entity#generic-properties

Reply:
Maybe two sensors, one that returns the current mode and one that returns the next scheduled datetime. The latter sensor should have device class timestamp.